pkg/minikube/process/process_test.go

```diff
@@ -0,0 +1,249 @@
+		}
+		if err != os.ErrProcessDone {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+	})
+
+	t.Run("terminate ignored", func(t *testing.T) {
+		cmd := startProcess(t, noterm)
+		defer cmd.Process.Kill()
+		err := process.Signal(cmd.Process.Pid, executable(noterm), syscall.SIGTERM)
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(200 * time.Millisecond)
+		exists, err := process.Exists(cmd.Process.Pid, executable(noterm))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !exists {
+			t.Fatalf("process terminated")
+		}
+	})
+
+	t.Run("kill", func(t *testing.T) {
+		cmd := startProcess(t, noterm)
+		defer cmd.Process.Kill()
+		err := process.Kill(cmd.Process.Pid, executable(noterm))
+		if err != nil {
+			t.Fatal(err)
+		}
+		waitForTermination(t, cmd)
+	})
+
+	t.Run("kill name mismatch", func(t *testing.T) {
+		cmd := startProcess(t, noterm)
+		defer cmd.Process.Kill()
+		err := process.Kill(cmd.Process.Pid, "no-such-process")
+		if err == nil {
+			t.Fatalf("Killed unrelated process")
+		}
+		if err != os.ErrProcessDone {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+	})
+}
+
+func startProcess(t *testing.T, cmd string) *exec.Cmd {
+	c := exec.Command(cmd)
+	stdout, err := c.StdoutPipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stdout.Close()
+
+	start := time.Now()
+	if err := c.Start(); err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Started process %q (pid=%v)", executable(cmd), c.Process.Pid)
+
+	// Synchronize with the process to ensure it set up signal handlers before
+	// we send a signal.
+	r := bufio.NewReader(stdout)
+	line, err := r.ReadString('\n')
+	if err != nil {
+		c.Process.Kill()
+		c.Wait()
+		t.Fatal(err)
+	}
+	if line != "READY\n" {
+		c.Process.Kill()
+		c.Wait()
+		t.Fatalf("Unexpected response: %q", line)
+	}
+	t.Logf("Process %q ready in %.6f seconds", executable(cmd), time.Since(start).Seconds())
+
+	// We must reap the process for testing termination. Otherwise it will
+	// become a zombie and Exists() will return true after termination.
+	go c.Wait()
+
+	return c
+}
+
+func waitForTermination(t *testing.T, cmd *exec.Cmd) {
+	timeout := 200 * time.Millisecond
+	start := time.Now()
+	for {
+		exists, err := process.Exists(cmd.Process.Pid, executable(cmd.Path))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !exists {
+			t.Logf("Process %q terminated in %.6f seconds", executable(cmd.Path), time.Since(start).Seconds())
+			break
+		}
+		if time.Since(start) > timeout {
+			t.Fatalf("Timeout waiting for %q", cmd)
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+}
+
+func build(t *testing.T, name string) (string, error) {
+	source := fmt.Sprintf("testdata/%s.go", name)
+
+	out := filepath.Join(t.TempDir(), name)
+	if runtime.GOOS == "windows" {
+		out += ".exe"
+	}
+
+	t.Logf("Building %q", name)
+	cmd := exec.Command("go", "build", "-o", out, source)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+	return out, nil
+}
+
+func executable(s string) string {
```

Comment:
Based on go-ps test we can use filepath.Base() in all cases.
https://github.com/mitchellh/go-ps/blob/ddafa7589c607e0e81a7436520ccdbac913665a2/process_test.go#L36